crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -0,0 +1,988 @@
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, BraintreeCancelResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        if item.response.errors.is_some() {
+            Ok(Self {
+                response: build_error_response(
+                    &item
+                        .response
+                        .errors
+                        .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?,
+                    item.http_code,
+                ),
+                ..item.data
+            })
+        } else {
+            let void_data = match &item.response.data {
+                Some(void_info) => void_info
+                    .reverse_transaction
+                    .as_ref()
+                    .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                    .reversal
+                    .as_ref(),
+                None => Err(errors::ConnectorError::ResponseDeserializationFailed)?,
+            };
+            let transaction_id = void_data
+                .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                .id
+                .clone();
+            Ok(Self {
+                status: enums::AttemptStatus::from(
+                    void_data
+                        .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                        .status
+                        .clone(),
+                ),
+                response: Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(
+                        transaction_id.to_string(),
+                    ),
+                    redirection_data: None,
+                    mandate_reference: None,
+                    connector_metadata: None,
+                    network_txn_id: None,
+                    connector_response_reference_id: None,
+                }),
+                ..item.data
+            })
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct BraintreePSyncRequest {
+    query: String,
+}
+
+impl TryFrom<&types::PaymentsSyncRouterData> for BraintreePSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsSyncRouterData) -> Result<Self, Self::Error> {
+        let transaction_id = item
+            .request
+            .connector_transaction_id
+            .get_connector_transaction_id()
+            .change_context(errors::ConnectorError::MissingConnectorTransactionID)?;
+        let query = format!("query {{ search {{ transactions(input: {{ id: {{is: \"{}\"}} }}, first: 1) {{ edges {{ node {{ id status createdAt amount {{ value currencyCode }} orderId }} }} }} }} }}", transaction_id);
```

Comment:
assign to constant

Reply:
It cannot be assigned to a constant as the query is dynamic, we have to insert `connector_transaction_id` in the query